### PR TITLE
Add files for the new st2garbagecollector service

### PR DESCRIPTION
**Pull request by Kami**
The title says it all.

On a related note - we need to add some developer docs on how to add a new service.

### CODE REVIEW

**Comment by Kami** [comment]
I looked over the existing files and I'm a bit confused in some places.

1. https://github.com/StackStorm/st2-packages/blob/master/packages/st2bundle/Makefile#L18

Any particular reason why we don't copy all the logging configs here? We only do it for st2auth, st2api, st2actions, but not for st2notifer, st2rulesengine, st2sensorcontainer, etc.

**Comment by Kami** [comment]
Please let me know if I missed some spot.

File: circle.yml [hunk 1]
```diff
@@ -59,7 +59,7 @@ test:
         parallel: true
   post:
     - .circle/docker.sh build st2bundle
-    - .circle/docker.sh build st2actionrunner st2api st2auth st2exporter st2notifier st2resultstracker st2rulesengine st2sensorcontainer
+    - .circle/docker.sh build st2actionrunner st2api st2auth st2exporter st2notifier st2resultstracker st2rulesengine st2sensorcontainer st2garbagecollector
```
Comment (armab) [inline circle.yml]: So it's new service/Docker container?
You need to add new Dockerfile here too: https://github.com/StackStorm/st2-dockerfiles

Also we need new repo in Docker hub (creating manually). I did it for you: https://hub.docker.com/r/stackstorm/st2garbagecollector/

Comment (Kami) [inline circle.yml]: Yeah, it's a new service which is part of  st2reactor package (for now).

I will have a look, we need to make sure all this gets documented :)

Comment (armab) [inline circle.yml]: Don't forget to include `st2garbagecollector` here: https://github.com/StackStorm/st2-packages/blob/master/circle.yml#L79

At the same time @Kami keep in mind that `docker build` and `docker push` commands are here temporarily, this logic should be moved to staging box eventually (eg. when all tests passed -> build and push Docker).

Comment (Kami) [inline circle.yml]: Good catch, added in 1df086ac093d982c4b833c936557d77aef7d5e19

**Comment by armab** [comment]
Regarding packaging @dennybaa please take a look

**Comment by armab** [comment]
@Kami Also one important thing I just remembered:

Everything that's included in any package should be duplicated in `st2bundle` package, see for example all init files are here: 
https://github.com/StackStorm/st2-packages/tree/master/packages/st2bundle/debian


`st2bundle` at the moment is used for building Docker.

**Comment by Kami** [comment]
@armab ah, I totally missed that.

Can we automate this - aka copy the files over in some make step or something like that? Copying it manually for sure means things will get out of sync at some point.

**Comment by dennybaa** [comment]
> I looked over the existing files and I'm a bit confused in some places.

> https://github.com/StackStorm/st2-packages/blob/master/packages/st2bundle/Makefile#L18
> Any particular reason why we don't copy all the logging configs here? We only do it for st2auth,  st2api, st2actions, but not for st2notifer, st2rulesengine, st2sensorcontainer, etc.

@Kami yeah exactly they are not copied here. makefiles used only for coping files which require renaming, standard `debian/install` handles the operation otherwise. Specifically for this it's here: https://github.com/StackStorm/st2-packages/blob/master/packages/st2bundle/debian/install#L6-L11

Also it's good to mention that rpm specs inherit debian code using helpers such as https://github.com/StackStorm/st2-packages/blob/master/rpmspec/helpers.spec#L20-L26.

**Comment by Kami** [comment]
@dennybaa That's for clarifying.

> Can we automate this - aka copy the files over in some make step or something like that? Copying it manually for sure means things will get out of sync at some point.

What do you think about that?

**Comment by dennybaa** [comment]
@Kami right now unfortunately nothing( I'll give a try tomorrow to think on it. It's a useful thing but we should concentrate on bugs fishing and fixing as of now.

**Comment by Kami** [comment]
OK, for now I will copy and paste it but lets get something better figured out soon :)

**Comment by Kami** [comment]
Copied the files over - should be all ready now, I think.

**Comment by Kami** [comment]
Build still failing - is there anything I need to do on my side? What's the process to merge it, do I need to merge st2-dockerfiles PR first?

**Comment by dennybaa** [comment]
Noop. Thank you @Kami, you've already added lots of useful stuff. Now I need to add rake build updates. Is it urgent?

**Comment by Kami** [comment]
Nope, not urgent - just wondering.

**Comment by dennybaa** [comment]
@Kami also verified for el6 https://circleci.com/gh/dennybaa/st2-packages/115, working. We can merge if no objections.

**Pull request merged**